test/websocket.js

```diff
@@ -0,0 +1,124 @@
+const assert = require("assert");
+const nodemailer = require("nodemailer");
+const request = require("supertest");
+const WebSocket = require("ws");
+
+const mailsacAPIKey = ""; // Generated by mailsac. See https://mailsac.com/api-keys
+const mailsacToAddress = ""; // Mailsac email address where the email will be sent
+const smtpUserName = ""; // Username for smtp server authentication
+const smtpPassword = ""; // Password for smtp server authentication
+const smtpHost = ""; // hostname of the smtp server
+const smtpPort = 587; // port the smtp is listening on
+
+let ws;
+
+describe("send email to mailsac", function () {
+  this.timeout(50000); // test can take a long time to run. This increases the default timeout for mocha
+
+  /* delete all messages in the inbox after the test runs to prevent leaky tests.
+       This requires the inbox to private, which is a paid feature of Mailsac.
+       The afterEach section could be omitted if using a public address
+    */
+  after(() =>
+    request("https://mailsac.com")
+      .delete(`/api/addresses/${mailsacToAddress}/messages`)
+      .set("Mailsac-Key", mailsacAPIKey)
+      .expect(204)
+  );
+  // close websocket after all tests finish
+  after(() => ws.close());
+
+  // Open websocket waiting for email. This websocket will be reused for tests in this file.
+  before(() => {
+    return new Promise((resolve, reject) => {
+      ws = new WebSocket(
+        `wss://sock.mailsac.com/incoming-messages?key=${mailsacAPIKey}&addresses=${mailsacToAddress}`
+      );
+      ws.on("message", (msg) => {
+        const wsMessage = JSON.parse(msg)
```

Comment:
Since `JSON.parse()` can throw, the test will hang if `msg` is not JSON for some reason. Best to wrap in a try/catch.